crates/lox_core/src/coords/states.rs

```diff
@@ -10,23 +10,34 @@ use float_eq::float_eq;
 use glam::{DMat3, DVec3};
 
 use crate::math::{mod_two_pi, normalize_two_pi};
-use crate::time::continuous::Time;
+use crate::time::continuous::{Time, TimeScale};
 
-pub trait TwoBodyState {
-    fn time(&self) -> Time;
-    fn to_cartesian_state(&self, grav_param: f64) -> CartesianState;
-    fn to_keplerian_state(&self, grav_param: f64) -> KeplerianState;
+pub trait TwoBodyState<T: TimeScale> {
```

Comment:
We should keep the structs in this module concrete. The `time` field is currently not needed and can be pushed up to the generic types.